plugins/inputs/docker/docker.go

```diff
@@ -296,6 +311,22 @@ func (d *Docker) gatherContainer(
 		tags[k] = label
 	}
 
+	// Add whitelisted environment variables to tags
+	if len(d.GatherEnvironment) > 0 {
+		info, err := inspectWrapper(d.client, ctx, container.ID)
+		if err != nil {
+			return fmt.Errorf("Error getting docker container inspect: %s", err.Error())
+		}
+
+		for _, envvar := range info.Config.Env {
+			kvs := strings.SplitN(envvar, "=", 2)
+			if !sliceContains(kvs[0], d.GatherEnvironment) {
+				continue
+			}
+			tags[kvs[0]] = kvs[1]
```

Comment:
Lets be extra careful and check the len of kvs, since SplitN could return a slice of length 1.